### ISSUE TITLE
exa -l fails because of file date

### ISSUE THREAD

**Issue by lccro** [ogham/exa#658]
```shell
$ RUST_BACKTRACE=full exa -l
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: SystemTimeError(6795364578.871345152s)', src/libcore/result.rs:1084:5
stack backtrace:
   0:        0x10ab92ced - std::panicking::default_hook::{{closure}}::hffc69ca72fa98b80
   1:        0x10ab927bc - std::panicking::rust_panic_with_hook::h058ef423beec40ce
   2:        0x10ab92f7d - std::panicking::continue_panic_fmt::h0bb2cdb45a553955
   3:        0x10ab93249 - rust_begin_unwind
   4:        0x10ac24dfa - core::panicking::panic_fmt::h2413d97d26203710
   5:        0x10ac25399 - core::result::unwrap_failed::hef33208e3f23e07a
   6:        0x10ab586bd - exa::output::table::Table::display::h7d61868c2e1aa281
   7:        0x10ab6b955 - <F as scoped_threadpool::FnBox>::call_box::hfeff2cdcb76e7a57
   8:        0x10ab83d6e - std::sys_common::backtrace::__rust_begin_short_backtrace::hf388fc36537a8fe9
   9:        0x10ab8489e - core::ops::function::FnOnce::call_once{{vtable.shim}}::hd49318dcebc5ef76
  10:        0x10ab9d13d - <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once::hf778f932793a5a82
  11:        0x10ab9d767 - std::sys::unix::thread::Thread::new::thread_start::ha4ccb7099fa6037f
  12:     0x7fff6f001109 - _pthread_start
[1]    35138 abort      RUST_BACKTRACE=full exa -l
```

Hope this helps:

```shell
$ stat public
  File: public/
  Size: 640             Blocks: 0          IO Block: 4096   directory
Device: 1000004h/16777220d      Inode: 34369309    Links: 20
Access: (0755/drwxr-xr-x)  Uid: (  501/     leo)   Gid: (   20/   staff)
Access: 1754-08-30 19:37:14.()/-,+/+. -0306
Modify: 1754-08-30 19:37:14.()/-,+/+. -0306
Change: 2020-02-29 20:26:46.219092665 -0300
 Birth: 2020-02-29 20:26:45.878952429 -0300
```

I have no idea how it got that date. After touching it solved the issue.

**Comment by azah** [comment]
I got the same error when running `ls` or `ls -la` on some statically generated files.

```
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: SystemTimeError(6795364578.871345152s)', /home/andrew/.cargo/registry/src/github.com-1ecc6299db9ec823/exa-0.9.0/src/fs/file.rs:331:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

```
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: SystemTimeError(6795364578.871345152s)', /home/andrew/.cargo/registry/src/github.com-1ecc6299db9ec823/exa-0.9.0/src/fs/file.rs:331:9
stack backtrace:
   0: backtrace::backtrace::libunwind::trace
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.40/src/backtrace/libunwind.rs:88
   1: backtrace::backtrace::trace_unsynchronized
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.40/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print_fmt
             at src/libstd/sys_common/backtrace.rs:77
   3: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
             at src/libstd/sys_common/backtrace.rs:59
   4: core::fmt::write
             at src/libcore/fmt/mod.rs:1052
   5: std::io::Write::write_fmt
             at src/libstd/io/mod.rs:1426
   6: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:62
   7: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:49
   8: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:204
   9: std::panicking::default_hook
             at src/libstd/panicking.rs:224
  10: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:472
  11: rust_begin_unwind
             at src/libstd/panicking.rs:380
  12: core::panicking::panic_fmt
             at src/libcore/panicking.rs:85
  13: core::option::expect_none_failed
             at src/libcore/option.rs:1199
  14: exa::fs::file::File::modified_time
  15: exa::output::table::Table::display
  16: <F as scoped_threadpool::FnBox>::call_box
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
zsh: abort      RUST_BACKTRACE=1 exa -la
```

**Comment by zzamboni** [comment]
I came looking for this, since I experienced the same problem, and for the same reason - I had a directory with a 1754 date created by Hugo. `touch`ing the directory solved the problem in the meantime, but it seems `hugo` insists on resetting the directory to an old date every time it runs.

**Comment by jgmtzmd** [comment]
same problem.

```
}jg@leap15 ~ $ exa -l
thread 'main' panicked at 'index out of bounds: the len is 0 but the index is 0', /home/abuild/rpmbuild/BUILD/rustc-1.32.0-src/src/libcore/slice/mod.rs:2463:10
stack backtrace:
   0: <unknown>
   1: <unknown>
   2: <unknown>
   3: <unknown>
   4: <unknown>
   5: <unknown>
   6: <unknown>
   7: <unknown>
   8: <unknown>
   9: <unknown>
  10: <unknown>
  11: <unknown>
  12: <unknown>
  13: __libc_start_main
  14: <unknown>
Aborted (core dumped)
```

this happens everytime i use the `-l` option.
My Os is opensuse 42.1, exa's version is 0.8.

**Comment by Freaky** [comment]
I made a fix for this in #670.  There's also #643, but I believe mine to be a better approach since it doesn't just paper-over the problem by clamping to the epoch.

@jgmtzmd That looks to be a different issue.  Can you repeat it without a stripped executable so you get a proper backtrace?

**Comment by buzuck** [comment]
Just for reference, the related Hugo issue (that might incite this other bug to be solved ^^' ).
https://github.com/gohugoio/hugo/issues/6161

**Comment by ogham** [comment]
The fix for this (#670) has been merged and will go out with the next release.
